### PR TITLE
all: update all docker images to use Go 1.16.2

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What

Update all docker images to use Go 1.16.2 and Ubuntu 18.04.

### Why

Go 1.14 is no longer supported or receiving security updates. Ubuntu needs upgrading to align the versions of Ubuntu in the runtime images with the build images.

### Known limitations

N/A

### Reviewing

If I have requested you review this PR it is because I believe you own or are the most knowledgeable person for a service impacted by this change. Please review the PR, and handle, or delegate someone else to handle, any deployment after merge for your respective service.

@accordeiro Ticker
@bartekn @ire-and-curses Horizon
@marcelosalloum Friendbot, Regulated Assets
@leighmcculloch Webauth, Recoverysigner, Keystore

### CODE REVIEW

**Review by accordeiro — APPROVED** [review]
The Ticker part looks good to me, thanks @leighmcculloch!

File: services/horizon/internal/scripts/check_release_hash/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Change to Go version used in CI or rebuild with --build-arg.
-ARG GO_IMAGE=golang:1.14.15-stretch
+ARG GO_IMAGE=golang:1.16.2-stretch
```
Comment (marcelosalloum) [inline services/horizon/internal/scripts/check_release_hash/Dockerfile:2]: ❓  Does it need to be a `-stretch` version?

**Review by marcelosalloum — APPROVED** [review]
The regulated assets reference server LGTM! I just added a question ❓  below.

Also, any reason to go with Ubuntu 18.04 rather than 20.04? Are you just playing safe there?

File: services/horizon/internal/scripts/check_release_hash/Dockerfile (hunk 1 above)
Comment (bartekn) [inline services/horizon/internal/scripts/check_release_hash/Dockerfile:2]:
```suggestion
ARG GO_IMAGE=golang:1.16.1-stretch
```
SDF's build server currently uses 1.16.1. To allow reproducible builds let's stick to 1.16.1.

File: services/horizon/docker/Dockerfile.dev [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.14-stretch AS builder
+FROM golang:1.16.2 AS builder
```
Comment (bartekn) [inline services/horizon/docker/Dockerfile.dev:1]:
```suggestion
FROM golang:1.16.1 AS builder
```
SDF's build server currently uses 1.16.1. To allow reproducible builds let's stick to 1.16.1. And thanks for fixing this, we forgot to update this file.

**Review by bartekn — APPROVED** [review]

File: services/horizon/internal/scripts/check_release_hash/Dockerfile (hunk 1 above)
Comment (leighmcculloch) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile:2]: Can we bump that to `.2`?

Comment (jacekn) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile:2]: Good point. Looking dockerhub tags it should be fine to switch to `golang:1.16.2` which defaults to buster

Comment (jacekn) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile:2]: @bartekn SDF Horizon builds are now Dockerized so they don't use go from the buildserver. We just need to bump image in the build job to match this

Comment (bartekn) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile:2]: We can bump it to `.2`. I'm just thinking that the next time we bump Golang version in our org we should create a meta issue with a list of places we need to update (right now I think it's Circle-CI, docker images, build server).

**Review by howardtw — APPROVED** [review]
keystore change lgtm. Thanks!

File: services/horizon/internal/scripts/check_release_hash/Dockerfile (hunk 1 above)
Comment (leighmcculloch) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile]: Thanks, fixed in ae7f8ab384ef5d824449a4d14b91e176362e703f.

Comment (leighmcculloch) [reply services/horizon/internal/scripts/check_release_hash/Dockerfile]: Resolving since we've updated to 1.16.1 now.

**Comment by leighmcculloch** [comment]
I realize the CircleCI yml versions are also behind and need updating too, and we still had the old 1.14 builds listed, which aren't needed, so I removed them too.

**Review by howardtw — APPROVED** [review]
CircleCI config update also looks good to me.

**Pull request merged**